rlog/2025-06-25-gsub-perf-imp-comparison.mdx

```diff
@@ -0,0 +1,392 @@
+A peer usually receives multiple copies of any transmitted message. 
+Excluding the first received message, all copies are duplicates. 
+We compute average duplicates received by a peer as $\frac{1}{N M} \sum_{j=1}^{M} \sum_{i=1}^{N} d_{i,j}$ 
+where $N$ and $M$ denote the number of peers and the number of transmitted messages, respectively, 
+and $d_{i,j}$ represents the number of duplicates received by peer $i$ for message $j$. 
+A similar mechanism computes average IWANT requests.
+
+
+Three simulation scenarios are considered: 
+
+- **Scenario 1:** The number of publishers and message size are kept constant while the network size gradually increases.
+
+- **Scenario 2:** The number of publishers and the network size remain constant while the message size gradually increases.
+ 
+- **Scenario 3:** The number of nodes and message size remain constant while the number of publishers gradually increases.
+
+In all experiments, we transmit multiple messages such that 
+every publisher sends exactly one message to the network. 
+After a publisher transmits a message, 
+each subsequent publisher waits for a specified interval (inter-message delay) before sending the next message. 
+
+Rotating publishers ensures that every message traverses a different path, 
+which helps achieve fair performance evaluation. 
+On the other hand, changing inter-message delays allows for creating varying traffic patterns. 
+A shorter inter-message delay implies more messages can be in transit simultaneously, 
+which helps evaluate performance against large message counts. 
+A longer delay ensures every message is fully disseminated before introducing a new message. 
+Similarly, increasing message size stresses the network. 
+As a result, we evaluate performance across a broader range of use cases. 
+
+
+The simulation details are presented in the tables below. 
+The experiments are conducted using the shadow simulator. 
+We uniformly set peer bandwidths and link latencies between 40-200 Mbps and 
+40-130 milliseconds in five variations. 
+
+
+**Table 1:** Simulation Scenarios.
+| **Experiment** | **No. of Nodes** | **No. of Publishers** | **Message Size (KB)** | **Inter-Message Delay (ms)** |
+|:-------------:|:-------------:|:---------:|:-------------:|:---------:|
+| Scenario 1 | 3000, 6000, 9000, 12000 | 7 | 150 | 10000 |
+| Scenario 2 | 1500 | 10 | 200, 600, 1000, 1400, 1800 | 10000 |
+| Scenario 3 | 1500 | 25, 50, 75, 100, 125 | 50 | 50 |
+
+
+**Table 2:** Simulation Parameters.
+| **Parameter** | **Value** | **Parameter** | **Value** |
+|:-------------:|:---------:|:-------------:|:---------:|
+| $D$ | 8 | $D_{low}$ | 6 |
+| $D_{lazy}$ | 6 | $D_{high}$ | 12 |
+| Gossip factor | 0.05 | Muxer | yamux |
+| Heartbeat interval | 1000 ms | Floodpublish | false |
+| Peer Bandwidth | 40-200 Mbps | Link Latency| 40-130ms |
+| $D_{announce}$ GossipSub v2.0| 7 | Forward delay in PPPT/v1.4 with delay| 35 ms |
+
+
+### Results

```

Comment:
In these charts, the interpretation seems to be left to the reader,  wdyt about giving a TLDR about what should we infer from these?